kernel/typeops.ml

```diff
@@ -69,6 +69,10 @@ let warn_bad_relevance ?loc x = warn_bad_relevance ?loc (Some x)
 
 let check_assumption env x t ty =
   let r = x.binder_relevance in
+  let () = match r with
+  | Sorts.Relevant | Sorts.Irrelevant -> ()
+  | Sorts.RelevanceVar _ -> anomaly Pp.(str "the kernel does not support sort variables")
+  in
```

Comment:
It indicates a problem in the upper layers, and as such is necessary to catch issues with relevance fixing. Since relevance fixing is going to be turned off by default anytime soon, I don't see why this shouldn't go first.

Reply:
Relevance being wrong is different from relevance being a variable. The first problem is a tactic issue, the second one a normalizing issue. The very reason I wrote this PR is that I need it to debug the mess when not fixing relevances, so it has to go in first.

Reply:
>The first problem is a tactic issue, the second one a normalizing issue.

OK that makes sense to me
Then you should switch the code order in type_of_case

Reply:
Done.